File: test/fixtures/java-rest-controllers/functional/quarkus-vertx-router.java
package com.example;

import io.vertx.ext.web.Router;
import jakarta.enterprise.context.ApplicationScoped;
import jakarta.enterprise.event.Observes;

@ApplicationScoped
public class VertxRoutes {
    void init(@Observes Router router) {
        router.get("/hello").handler(rc -> rc.response().end("ok"));
        router.post("/items").handler(rc -> rc.response().end("created"));
    }
}

File: test/fixtures/kotlin-rest-controllers/functional/quarkus-vertx-router.kt
package com.example

import io.vertx.ext.web.Router
import jakarta.enterprise.context.ApplicationScoped
import jakarta.enterprise.event.Observes

@ApplicationScoped
class VertxRoutes {
    fun init(@Observes router: Router) {
        router.get("/hello").handler { rc -> rc.response().end("ok") }
        router.post("/items").handler { rc -> rc.response().end("created") }
    }
}
